test/immutable.js

```diff
@@ -151,8 +151,9 @@ describe('Immutable Data', () => {
         )
       .then(() => app.mutableData.newPublic(testXorName, TYPE_TAG))
       .then((md) => md.getEntries())
-      .then((entries) => entries.forEach((key, value) => {
-        app.immutableData.fetch(value.buf)
+      .then((entries) => entries.listEntries())
+      .then((entriesArray) => entriesArray.forEach((entry) => {
```

Comment:
I think we are missing something here, shouldn't we be doing all the `fetch`'s within a Promise.all?